mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -192,41 +293,159 @@ impl TendermintCoin {
 
         Ok(TendermintCoin(Arc::new(TendermintCoinImpl {
             ticker,
-            rpc_client,
+            rpc_clients,
             account_id,
             account_prefix: protocol_info.account_prefix,
             priv_key: priv_key.to_vec(),
             decimals: protocol_info.decimals,
             denom,
             chain_id,
+            gas_price: protocol_info.gas_price,
+            avg_block_time,
             sequence_lock: AsyncMutex::new(()),
+            tokens_info: Mutex::new(HashMap::new()),
         })))
     }
 
-    async fn my_account_info(&self) -> MmResult<BaseAccount, TendermintCoinRpcError> {
-        let path = AbciPath::from_str("/cosmos.auth.v1beta1.Query/Account").expect("valid path");
+    // TODO
+    // Save one working client to the coin context, only try others once it doesn't
+    // work anymore.
+    // Also, try couple times more on health check errors.
+    async fn rpc_client(&self) -> MmResult<HttpClient, TendermintCoinRpcError> {
+        for rpc_client in self.rpc_clients.iter() {
+            match rpc_client.perform(HealthRequest).timeout(Duration::from_secs(3)).await {
+                Ok(res) => match res {
+                    Ok(_) => return Ok(rpc_client.clone()),
+                    Err(e) => {
+                        log::warn!(
+                            "Recieved error from Tendermint rpc node during health check. Error: {:?}",
+                            e
+                        );
+                    },
+                },
+                Err(_) => {
+                    log::warn!("Tendermint rpc node: {:?} got timeout during health check", rpc_client);
+                },
+            };
+        }
+
+        MmError::err(TendermintCoinRpcError::PerformError(
+            "All the current rpc nodes are unavailable.".to_string(),
+        ))
+    }
+
+    #[inline(always)]
+    fn gas_price(&self) -> f64 { self.gas_price.unwrap_or(DEFAULT_GAS_PRICE) }
+
+    #[allow(unused)]
+    async fn get_latest_block(&self) -> MmResult<GetLatestBlockResponse, TendermintCoinRpcError> {
+        let path = AbciPath::from_str(ABCI_GET_LATEST_BLOCK_PATH).expect("valid path");
+
+        let request = GetLatestBlockRequest {};
+        let request = AbciRequest::new(Some(path), request.encode_to_vec(), None, false);
+
+        let response = self.rpc_client().await?.perform(request).await?;
+
+        Ok(GetLatestBlockResponse::decode(response.response.value.as_slice())?)
+    }
+
+    #[allow(unused)]
+    async fn get_block_by_height(&self, height: i64) -> MmResult<GetBlockByHeightResponse, TendermintCoinRpcError> {
+        let path = AbciPath::from_str(ABCI_GET_BLOCK_BY_HEIGHT_PATH).expect("valid path");
+
+        let request = GetBlockByHeightRequest { height };
+        let request = AbciRequest::new(Some(path), request.encode_to_vec(), None, false);
```

Comment:
Could you please add `None` and `false` constants? They're passed into `AbciRequest` multiple times.

Reply:
Done